test/model.js

```diff
@@ -1445,4 +1445,40 @@
     assert.equal(model.id, 3);
   });
 
+  QUnit.test('#4034 - Model.prototype.update()', function(assert) {
+    var model = new Backbone.Model({prop1: 'value1'});
+    model.update('prop1', function(prop1) {
+      assert.equal(this, model);
+      return prop1 + '-changed';
+    });
+    assert.strictEqual(model.get('prop1'), 'value1-changed');
+  });
+
+  QUnit.test('#4034 - Model.prototype.update() with multiple attributes', function(assert) {
+    var model = new Backbone.Model({
+      prop0: 'value0',
+      prop1: 'value1'
+    });
+    model.update(['prop0', 'prop1'], function(prop0, prop1) {
```

Comment:
Personal taste maybe, but I find @benmccormick proposition for multiple args nicer, e.g:

```js
model.update(['totalCount', "team1Count"], count => count + 1);
```